src/app/modules/issue/issue.service.ts

```diff
@@ -0,0 +1,182 @@
+
+const logger = createLoggerWithLabel(module)
+
+const FormIssueModel = getFormIssueModel(mongoose)
+
+/**
+ * Inserts given form issue to the database.
+ * @param formId the formId to insert the feedback for
+ * @param issue the description of the issue reported
+ * @param email the contact info for form creator to follow up the issue with
+ * @returns ok(true) if successfully inserted
+ * @returns err(DatabaseError) on database error
+ */
+export const insertFormIssue = ({
+  formId,
+  issue,
+  email,
+}: {
+  formId: string
+  issue: string
+  email?: string
+}): ResultAsync<IFormIssueSchema, FormNotFoundError | DatabaseError> => {
+  if (!mongoose.Types.ObjectId.isValid(formId)) {
+    return errAsync(new FormNotFoundError())
+  }
+  return ResultAsync.fromPromise(
+    FormIssueModel.create({
+      formId: formId,
+      issue: issue,
+      ...(email && { email: email }),
+    }),
+    (error) => {
+      logger.error({
+        message: 'Database error when creating form issue document',
+        meta: {
+          action: 'insertFormIssue',
+          formId,
+        },
+        error,
+      })
+
+      return new DatabaseError('Form issue could not be created')
+    },
+  )
+}
+
+/**
+ * Checks if there are any prior issues reported today given a formId.
+ * The `_id` field of the form is used to check for prior issues, even though
+ * mongo objectId is accurate only up to seconds, it should be fine, as we do
+ * not expect more than 1 issue complain per form per second. `created` field is
+ * not used for now as the column is not being indexed (to reduce additional
+ * write overhead).
+ * In the unlikely chance of this happening, form admin and collaborators may
+ * receive up to the number of issues reported per second per form.
+ * @param formIssue the formIssue object that was just inserted
+ * @returns ok(true) if successfully inserted
+ * @returns err(DatabaseError) on database error
+ */
+const getIsFirstIssueForFormToday = ({
+  formIssue,
+}: {
+  formIssue: IFormIssueSchema
+}): ResultAsync<boolean, DatabaseError> => {
+  const startOfDay =
+    formIssue._id instanceof ObjectId
+      ? getStartOfDay({ date: formIssue._id.getTimestamp() })
+      : getStartOfDay()
+
+  const issueId = formIssue._id as ObjectId
+  const queryFilter = {
+    formId: formIssue.formId,
+    _id: {
+      $gte: ObjectId.createFromTime(startOfDay.getTime() / 1000),
+      $lt: issueId,
+    },
+  }
+  return ResultAsync.fromPromise(
+    FormIssueModel.countDocuments(queryFilter).limit(1).exec(),
+    (error) => {
+      logger.error({
+        message: 'Database error when creating form issue document',
+        meta: {
+          formId: formIssue.formId,
+          action: 'insertFormIssue',
+        },
+        error,
+      })
+      return new DatabaseError(
+        'Failed to count the document in the formIssue collection',
+      )
+    },
+  ).andThen((count) => {
+    return okAsync(count == 0)
```

Comment:
Missed out another `=`? 

We usually still use strict equality (`===`) even though the operand are both of the same type. This is to prevent potential pitfall when the operands changes.

Reply:
Ahh.. thanks for pointing it out, not used to typescript/javascript.